Recent failures with: TypeError: Cannot read properties of undefined (reading 'title')

Hi,

Recently we have started experiencing an issue with running `md-to-notion` and we are trying to debug it. Error we get is below. Doesn't matter if it is on version `0.5.0` or previous one (`0.4.0` which worked for us and we tested in now and also fails). We have also tested on a fresh/blank Notion page. We are not an experts in nodeJS/JS/TS etc. so I thought maybe this would ring a bell to maintainers and to point us in some direction. We keep investigating but any input would be helpful. We run it on GitHub Actions and had no issues few weeks ago, not sure yet what have changed since then 🤷 

Thank you.

```
npm warn exec The following package was not found and will be installed: @vrerv/md-to-notion@0.5.0
/home/runner/.npm/_npx/e5b811e6ecb666d5/node_modules/@vrerv/md-to-notion/build/src/sync-to-notion.js:22
    return pageResponse.properties.title.title[0].text.content;
                                         ^

TypeError: Cannot read properties of undefined (reading 'title')
    at getPageTitle (/home/runner/.npm/_npx/e5b811e6ecb666d5/node_modules/@vrerv/md-to-notion/build/src/sync-to-notion.js:22:42)
    at collectPages (/home/runner/.npm/_npx/e5b811e6ecb666d5/node_modules/@vrerv/md-to-notion/build/src/sync-to-notion.js:56:27)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async collectCurrentFiles (/home/runner/.npm/_npx/e5b811e6ecb666d5/node_modules/@vrerv/md-to-notion/build/src/sync-to-notion.js:69:5)
    at async Command.main (/home/runner/.npm/_npx/e5b811e6ecb666d5/node_modules/@vrerv/md-to-notion/build/src/md-to-notion-cli.js:36:25)

Node.js v18.20.5
Error: Process completed with exit code 1.
```

--- a/src/sync-to-notion.ts
+++ b/src/sync-to-notion.ts
@@ -36,7 +36,16 @@ function commonPageKey(parentId: string, title: string): string {
 function getPageTitle(pageResponse: GetPageResponse): string {
   // eslint-disable-next-line @typescript-eslint/ban-ts-comment
   // @ts-ignore
-  return pageResponse.properties.title.title[0].text.content
+  const titles = pageResponse.properties.title.title
+  if (titles && titles.length > 0) {
+    return titles[0].plain_text
+  }
+  logger(LogLevel.ERROR, "Error no title found", { pageResponse })
+  throw new Error(
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    `No title found. Please set a title for the page ${pageResponse.url} and try again.`
+  )
 }
 
 function newNotionPageLink(response: PageObjectResponse): NotionPageLink {

--- a/test/sync-to-notion.test.ts
+++ b/test/sync-to-notion.test.ts
@@ -1,4 +1,5 @@
 import {
+  collectCurrentFiles,
   findMaxDepth,
   NotionPageLink,
   syncToNotion,
@@ -425,5 +426,39 @@ describe("syncToNotion", () => {
     expect(linkMap.size).toBe(2)
     expect(linkMap.get(`./subfolder`)?.id).toBe("subfolder")
     expect(linkMap.get(`./subfolder/subfile`)?.id).toBe("subfile")
+  })
+})
+
+describe("collectCurrentFiles", () => {
+  const mockNotionClient = new Client({ auth: "test-token" })
+  //mock call to Notion API and return no title page response
+  mockNotionClient.pages.retrieve = jest.fn().mockImplementation(async () => {
+    return {
+      object: "page",
+      id: "page-1",
+      url: "https://vrerv.com/page-1",
+      properties: {
+        title: {
+          title: [],
+        },
+      },
+    }
+  })
+
+  it("should throw error if there's no page title page", async () => {
+    const pageId = "test-page-id"
+
+    try {
+      await collectCurrentFiles(mockNotionClient, pageId)
+      fail("Should throw error if there's no page title page")
+    } catch (e: unknown) {
+      if (e instanceof Error) {
+        expect(e.message).toBe(
+          "No title found. Please set a title for the page https://vrerv.com/page-1 and try again."
+        )
+      } else {
+        fail("Should throw error if there's no page title page")
+      }
+    }
   })
 })

